numpy/lib/function_base.py

```diff
@@ -671,11 +672,20 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
         mn -= 0.5
         mx += 0.5
 
+    # density overrides the normed keyword
+    if density is not None:
+        normed = False
+
+    # parse the overloaded bins argument
+    n_equal_bins = None
+    bin_edges = None
```

Comment:
Using these names makes things a lot clearer, rather than `bins` meaning different things in different places